students/TimPauley/session02/code/charges_calc_update.py

```diff
@@ -0,0 +1,157 @@
+    """
+    format_logger = '%(asctime)s %(filename)s:%(lineno)-3d%(levelname)s'
+    format_logger += ' %(message)s'
+    log_file = datetime.datetime.now().strptime('%Y-%m-%d') + '.log'
+    debug_leve = int(debug_level)
+    if debug_level == 0:
+        logging.basicConfig(level=logging.CRITICAL)
+    if debug_level > 0:
+        logging.basicConfig(format=format_logger, level=logging.ERROR)
+        f_handler = logging.FileHandler(log_file)
+        f_handler.setLevel(logging.WARNING)
+        logging.getLogger().addHandler(f_handler)
+    if debug_level == 2:
+        logging.getLogger().setLevel(logging.WARNING)
+    if debug_level == 3:
+        logging.getLogger().setLevel(logging.DEBUG)        
+
+def load_rentals_file(filename):
+    """
+    Loads the json file of rental data
+    """
+    logging.debug("load_rentals_file")
+    logging.debug("Opens json file and readz it in")
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+        except:
+            logging.error(error_msg)
+            logging.debug(i)
+    return data
+
+def calculate_additional_fields(data):
+    """
+    Calculates additional properties
+    """
+    logging.debug("calculate_additional_fields")
+    if data_new:
+        for value in data_new.values():
+            try:
+                warning_string = f"Incorrect date for {value['product_code']}"
+                if value["rental_start"] == '' or value["rental_end"] == '':
+                    rental_start = 0
+                    rental_end = rental_start
+                    logging.warning(warning_string)
+                    logging.debug(value)
+                else:
+                    rental_start = datetime.datetime.strptime(
+                        value['rental_start'], '%m/%d/%y')
+                    rental_end = datetime.datetime.strptime(
+                        value['rental_end'], '%m/%d/%y')
+                if rental_start >= rental_end:
+                    value['total_days'] = 0
+                    logging.warning(warning_string)
+                    logging.debug(value)
+                else:
+                    value['total_days'] = (rental_end - rental_start).days
+                value['total_price'] = value['total_days'] * value[
+                    'price_per_day']
+                value['sqrt_total_price'] = math.sqrt(value['total_price'])
+                if value["total_price"] == 0:
+                    value["unit_cost"] = 0
+                    warning_price = "Incorrect price for "
+                    warning_price += f"{value['product_code']}"
+                    logging.warning(warning_price)
+                    logging.debug(value)
+                else:
+                    value['unit_cost'] = value['total_price'] / value[
+                        'units_rented']
+            except Exception as error_msg:
+                logging.error(error_msg)
+                logging.debug(value)
+                data_new = None
```

Comment:
You have some good error handling in this block. Keep in mind, you really want to limit what is in a try block to just the things you expect to go wrong. Try blocks should have max 3-4 lines in them.